File: examples/rust/arithmetic/Arithmetic.runar.rs
use runar::prelude::*;

#[runar::contract]
struct Arithmetic {
    #[readonly]
    target: Int,
}

impl Arithmetic {
    pub fn verify(&self, a: Int, b: Int) {
        let sum = a + b;
        let diff = a - b;
        let prod = a * b;
        let quot = a / b;
        let result = sum + diff + prod + quot;
        assert!(result == self.target);
    }
}

File: examples/go/arithmetic/Arithmetic.runar.go
package contract

import runar "github.com/icellan/runar/packages/runar-go"

type Arithmetic struct {
	runar.SmartContract
	Target runar.Int `runar:"readonly"`
}

func (c *Arithmetic) Verify(a runar.Int, b runar.Int) {
	sum := a + b
	diff := a - b
	prod := a * b
	quot := a / b
	result := sum + diff + prod + quot
	runar.Assert(result == c.Target)
}
